text/0000-edition-2021.md

```diff
@@ -0,0 +1,226 @@
+
+[RFC 2229]: https://github.com/rust-lang/rfcs/pull/2229
+
+## Access to the 2021 edition on nightly
+
+The nightly compiler will make a 2021 edition available in an unstable form. This will allow us to build and test the migrations for various features that will be part of the 2021 edition. 
+
+**Warning:** It is not recommended for crates, even nightly-only crates, to adopt the 2021 Edition too early. Like any unstable feature, the future editions can change without notice and crates are expected to keep up. Furthermore, the automatic migrations are designed to be used exactly once; if a crate C is migrated from Edition X to Edition Y, and then new features or migrations are added to Edition Y, the crate C may not be able to access those new migrations and the changes may have to be done manually.
+
+## Advertising and documenting the edition
+
+The expectation is that the 2021 Edition will not be marketed as an "all encompassing event" in the same way as the 2018 Edition.
+
+Producing an edition requires producing the following key artifacts. This list is not meant to be exhaustive; as part of executing on the 2021 Edition, we will be producing more detailed guidelines for how the process works that can be used as a template for future editions if desired.
+
+* "What is an edition" page on the main Rust web site
+* Edition migration guide
+* Edition status tracking page, likely part of a general tracking page as envisioned by [RFC 3037](https://github.com/rust-lang/rfcs/pull/3037).
+* Blog post announcing the edition -- this will be the "highlight" part of the standard release blog post
+
+## Metrics for a successful edition
+
+The following metrics are an attempt to quantify some of the goals we have for editions.
+
+### Easy migration
+
+* 90% of crater-seen crates migrate without manual intervention.
+    * This can be measured automatically, subject to the limitations of crater (e.g., linux only).
+* 90% of crates that were migrated to the newer edition could be migrated in less than 1 hour
+    * This includes both rustfix but also manual migraton.
+    * This will require a survey or other forms of self-reporting.
+
+### Adoption
+
+* 75% of Rust survey respondents in 2022 indicate that they are using Rust 2021
+    * If we don't see this number, it may not be a problem, but it'd be worth investigating why.
+
+# Drawbacks
+[drawbacks]: #drawbacks
+
+Executing an edition requires coordination. There has also been some concern that Rust is making too many changes and moving too quickly, and releasing a new edition could feed those narratives. On the other hand, the fact that this edition is relatively limited in scope and that it will be marketed less aggressively also helps here. Further, continuing the regular cadence for editions has its own advantages, and helps us to make some changes (such as closure capture or format printing) that are very valuable.
```

Comment:
```suggestion
Executing an edition requires coordination. There has also been some concern that Rust is making too many changes and moving too quickly, and releasing a new edition could feed those fears. On the other hand, the fact that this edition is relatively limited in scope and that it will be marketed less aggressively also helps here. Further, continuing the regular cadence for editions has its own advantages, and helps us to make some changes (such as closure capture or format printing) that are very valuable.
```